File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/INotificationsApi.java
package dev.ragnarok.fenrir.api.interfaces;

import androidx.annotation.CheckResult;

import dev.ragnarok.fenrir.api.model.response.NotificationsResponse;
import dev.ragnarok.fenrir.model.AnswerVKOfficialList;
import io.reactivex.rxjava3.core.Single;


public interface INotificationsApi {

    @CheckResult
    Single<Integer> markAsViewed();

    @CheckResult
    Single<NotificationsResponse> get(Integer count, String startFrom, String filters,
                                      Long startTime, Long endTime);

    @CheckResult
    Single<AnswerVKOfficialList> getOfficial(Integer count, Integer startFrom, String filters, Long startTime, Long endTime);

}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/interfaces/INotificationsApi.kt
package dev.ragnarok.fenrir.api.interfaces

import androidx.annotation.CheckResult
import dev.ragnarok.fenrir.api.model.response.NotificationsResponse
import dev.ragnarok.fenrir.model.AnswerVKOfficialList
import io.reactivex.rxjava3.core.Single

interface INotificationsApi {
    @CheckResult
    fun markAsViewed(): Single<Int>

    @CheckResult
    operator fun get(
        count: Int?, startFrom: String?, filters: String?,
        startTime: Long?, endTime: Long?
    ): Single<NotificationsResponse>

    @CheckResult
    fun getOfficial(
        count: Int?,
        startFrom: Int?,
        filters: String?,
        startTime: Long?,
        endTime: Long?
    ): Single<AnswerVKOfficialList>

    @CheckResult
    fun hide(query: String?): Single<Int>
}